infrastructure/irsa.tf

```diff
@@ -0,0 +1,31 @@
+module "vpc_cni_irsa_role" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+
+  role_name = "stefanb-vpc-cni"
+
+  attach_vpc_cni_policy = true
+
+
+  vpc_cni_enable_ipv4 = true
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.iam_iam-github-oidc-provider.arn
+      namespace_service_accounts = ["default:my-app", "canary:my-app"]
+    }
+  }
+}
+
+module "irsa_load_balancer" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  role_name = "task2-irsa-for-LB"
+
+  attach_load_balancer_controller_policy = true
+
+  oidc_providers = {
+    one = {
+      provider_arn               = module.iam_iam-github-oidc-provider.arn
+      namespace_service_accounts = ["vegait-load-balancer:load-balancer"]
```

Comment:
make sure you remember namespace and service account names that you put here as they need to be 1 on 1 with the ones that you will put in helm chart when you want to install lb controller.